memgpt/agent.py

```diff
@@ -522,6 +499,7 @@ def handle_ai_response(self, response_message):
             # Failure case 3: function failed during execution
             self.interface.function_message(f"Running {function_name}({function_args})")
             try:
+                function_args["self"] = self  # need to attach self to arg since it's dynamically linked

```

Comment:
We could potentially change this to be `agent` instead of `self` in the required function signature. `self` is less likely to have a naming collision though.